packages/contracts-bedrock/test/setup/Setup.sol

```diff
@@ -0,0 +1,201 @@
+            hex"7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3"
+        );
+
+        Deploy.run();
+
+        optimismPortal = OptimismPortal(mustGetAddress("OptimismPortalProxy"));
+        l2OutputOracle = L2OutputOracle(mustGetAddress("L2OutputOracleProxy"));
+        systemConfig = SystemConfig(mustGetAddress("SystemConfigProxy"));
+        l1StandardBridge = L1StandardBridge(mustGetAddress("L1StandardBridgeProxy"));
+        l1CrossDomainMessenger = L1CrossDomainMessenger(mustGetAddress("L1CrossDomainMessengerProxy"));
+        addressManager = AddressManager(mustGetAddress("AddressManager"));
+        l1ERC721Bridge = L1ERC721Bridge(mustGetAddress("L1ERC721BridgeProxy"));
+        l1OptimismMintableERC20Factory =
+            OptimismMintableERC20Factory(mustGetAddress("OptimismMintableERC20FactoryProxy"));
+        protocolVersions = ProtocolVersions(mustGetAddress("ProtocolVersionsProxy"));
+
+        vm.label(address(l2OutputOracle), "L2OutputOracle");
+        vm.label(mustGetAddress("L2OutputOracleProxy"), "L2OutputOracleProxy");
+        vm.label(address(optimismPortal), "OptimismPortal");
+        vm.label(mustGetAddress("OptimismPortalProxy"), "OptimismPortalProxy");
+        vm.label(address(systemConfig), "SystemConfig");
+        vm.label(mustGetAddress("SystemConfigProxy"), "SystemConfigProxy");
+        vm.label(address(l1StandardBridge), "L1StandardBridge");
+        vm.label(mustGetAddress("L1StandardBridgeProxy"), "L1StandardBridgeProxy");
+        vm.label(address(l1CrossDomainMessenger), "L1CrossDomainMessenger");
+        vm.label(mustGetAddress("L1CrossDomainMessengerProxy"), "L1CrossDomainMessengerProxy");
+        vm.label(address(addressManager), "AddressManager");
+        vm.label(address(l1ERC721Bridge), "L1ERC721Bridge");
+        vm.label(mustGetAddress("L1ERC721BridgeProxy"), "L1ERC721BridgeProxy");
+        vm.label(address(l1OptimismMintableERC20Factory), "OptimismMintableERC20Factory");
+        vm.label(mustGetAddress("OptimismMintableERC20FactoryProxy"), "OptimismMintableERC20FactoryProxy");
+        vm.label(address(protocolVersions), "ProtocolVersions");
+        vm.label(mustGetAddress("ProtocolVersionsProxy"), "ProtocolVersionsProxy");
+        vm.label(AddressAliasHelper.applyL1ToL2Alias(address(l1CrossDomainMessenger)), "L1CrossDomainMessenger_aliased");
+    }
+
+    /// @dev Sets up the L2 contracts. Depends on `L1()` being called first.
+    function L2(DeployConfig cfg) public {
+        // Set up L2. There are currently no proxies set in the L2 initialization.
+        vm.etch(
+            Predeploys.L2_CROSS_DOMAIN_MESSENGER,
+            address(new L2CrossDomainMessenger(address(l1CrossDomainMessenger))).code
+        );
+        l2CrossDomainMessenger = L2CrossDomainMessenger(payable(Predeploys.L2_CROSS_DOMAIN_MESSENGER));
+        l2CrossDomainMessenger.initialize();
```

Comment:
Combined with setting these known variables as constants [above](https://github.com/ethereum-optimism/optimism/pull/7928/files#r1383861960), IMO this is easier to read, and can cut out a lot of noise here.

```suggestion
        vm.etch(
            address(l2CrossDomainMessenger), address(new L2CrossDomainMessenger(address(l1CrossDomainMessenger))).code
        );
        l2CrossDomainMessenger.initialize();
```